pkg/watcher/internal/metricsprovider/prometheus.go

```diff
@@ -61,10 +72,70 @@ func NewPromClient(opts watcher.MetricsProviderOpts) (watcher.MetricsProviderCli
 		promAddress = opts.Address
 	}
 
+	// Ignore TLS verify errors if InsecureSkipVerify is set
+	roundTripper := api.DefaultRoundTripper
+	if opts.EnableOpenShiftAuth{
+		// Create the config for kubernetes client
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			// Get the kubeconfig path
+			kubeConfigPath, ok :=  os.LookupEnv(kubeConfig)
+			if !ok {
+				kubeConfigPath = defaultKubeConfig
+			}
+			config, err = clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+        }
```

Comment:
Indentation seems incorrect here and below.

Reply:
Will fix this issue.